python/pylibcudf/pylibcudf/interop.pyx

```diff
@@ -27,6 +25,14 @@ from .types cimport DataType, type_id
 from .types import LIBCUDF_TO_ARROW_TYPES
 from ._interop_helpers import ColumnMetadata
 
+try:
+    import pyarrow as pa
+    pa_err = None
+except ImportError as e:
+    pa = None
+    pa_err = e
```

Comment:
It will be surprising to users if, when calling other functions like `from_arrow`, they see an `ImportError`. I suggest that in addition to saving this error we create another `RuntimeError` (pick a better type if you can think of one) that says "pyarrow was not available on your system" or so. You can then `raise new_error from pa_err` that way the user sees the more helpful error first but the import issue is also surfaced.